### ISSUE TITLE
Add validation of OrgID in GetDashboard method

### ISSUE THREAD

**Issue by suntala** [grafana/grafana#73064]
Follow-up on [this PR comment](https://github.com/grafana/grafana/pull/65184#discussion_r1149829188) regarding the validation of OrgID. This change is needed because we rely on OrgID for uniqueness. 

Some suggestions that came up from a [daily discussion](https://docs.google.com/document/d/1MwUr6u3ABzqESNfh_beWW08xTDmyHtQpPIT2LnKiKK0/edit) on March 28th, 2023:

- Either orgId through rewriting to not using xorm magic
- Make sure xorm always sets it
- Change it here
- Basically
  - Not allowing it to be zero
  - Allowing it to be zero
